src/schema/organization/contributors/resolvers.js

```diff
@@ -1,61 +1,131 @@
 import fetchData from '../../../github/dataFetcher';
 import githubQueries from '../../../github/queries';
-import { GeneralSet, getOldestDate } from '../../../utils'
+import { getOldestDate } from '../../../utils';
+
+/**
+ * @class A variant of Set, which the difference is on the equality condition of objects
+ */
+export class ContributorSet {
+  constructor() {
+      this.map = new Map();
+      this[Symbol.iterator] = this.values;
+  }
+
+  add(item) {
+    this.map.set(item.login, item);
+  }
+
+  values() {
+      return this.map.values();
+  }
+
+  delete(item) {
+      return this.map.delete(item.login);
+  }
+}
 
 /**
  * Process the repositories array to leave only the information the schema needs.
  * @param {Array} repositories 
  * @return processed repositories array
  */
-const process = (repositories) => {
-  const contributors = new GeneralSet();
+const processContributors = (repositories) => {
+  const contributors = new ContributorSet();
 
   repositories.forEach(repository => {
     repository.mentionableUsers.nodes.forEach(user => {
-      const firstPullRequest = user.contributionsCollection.pullRequestContributions.nodes[0];
-      const firstIssue = user.contributionsCollection.issueContributions.nodes[0];
-      const firstReview = user.contributionsCollection.pullRequestReviewContributions.nodes[0];
-
-      let currentDate = undefined;
-
-      if (firstPullRequest || firstIssue || firstReview) {
-        const firstContributions = [firstPullRequest, firstIssue, firstReview];
-        const dates = [];
-        
-        firstContributions.forEach(contribution => {
-          if (contribution) dates.push(contribution.firstContributionDate);
-        });
-
-        currentDate = getOldestDate(dates);
-      }
-
-      user.contributionsCollection = undefined;
-      user.firstContributionDate = currentDate;
+      user.a = 1;
       contributors.add(user);
     });
   });
   
   return Array.from(contributors.values());
 };
 
+/**
+ * Process the user object and returns the Date of the first contribution or undefined
+ * if the user has no contributions.
+ * @param {Object} user 
+ * @return {Date} first contribution date
+ */
+const processFirstContributionDate = (user) => {
+  if (user) {
+    const issuesNodes = user.contributionsCollection.issueContributions.nodes;
+    const pullRequestsNodes = user.contributionsCollection.pullRequestContributions.nodes;
+    const reviewsNodes = user.contributionsCollection.pullRequestReviewContributions.nodes;
+    const contributions = [issuesNodes, pullRequestsNodes, reviewsNodes];
+
+    const dates = [];
+    var firstContrib;
+    contributions.forEach(contribution => {
+      if (contribution && contribution.length > 0) {
+        firstContrib = contribution[0].occurredAt;
+        dates.push(new Date(firstContrib)); 
+      }
+    });
+
+    var result = undefined;
+    if (dates.length > 0)
+      result = getOldestDate(dates);
```

Comment:
Change this for:
```suggestion
  return dates.length > 0 && getOldestDate(dates);
```
Because, if dates.length > 0  is true, the other expression is evaluated and if true, return it.

Reply:
but if any of the expressions is false, the result will be false not undefined.

Reply:
Yes, you do this, setting result = undefined in previous line

Reply:
Same behaviour